src/RealtimeEffectPanel.cpp

```diff
@@ -0,0 +1,796 @@
+         mChangeButton->SetMinSize(mEnableButton->GetMinSize());
+         mChangeButton->Bind(wxEVT_BUTTON, &RealtimeEffectControl::OnChangeButtonClicked, this);
+
+         //Show effect settings
+         auto optionsButton = safenew ThemedButtonWrapper<wxButton>(this, wxID_ANY, wxT(""), wxDefaultPosition, wxDefaultSize, wxNO_BORDER);
+         optionsButton->SetBitmapLabelIndex(bmpEffectSettingsNormal);
+         optionsButton->SetBitmapPressedIndex(bmpEffectSettingsDown);
+         optionsButton->SetBitmapCurrentIndex(bmpEffectSettingsHover);
+         optionsButton->SetBackgroundColorIndex(clrEffectListItemBackground);
+         optionsButton->SetMinSize(optionsButton->GetBitmapLabel().GetSize());
+         optionsButton->Bind(wxEVT_BUTTON, &RealtimeEffectControl::OnOptionsClicked, this);
+         
+         auto dragArea = safenew wxStaticBitmap(this, wxID_ANY, theTheme.Bitmap(bmpDragArea));
+         dragArea->Disable();
+         sizer->Add(
+            dragArea,
+            wxSizerFlags().Center().Border(wxLEFT, 5)
+         );
+         sizer->Add(
+            mEnableButton,
+            wxSizerFlags().Center().Border(wxLEFT, 5)
+         );
+         sizer->Add(
+            mChangeButton,
+            wxSizerFlags().Center().Border(wxLEFT, 5).Proportion(1)
+         );
+         sizer->Add(
+            optionsButton,
+            wxSizerFlags().Center().Border(wxLEFT | wxRIGHT, 5)
+         );
+
+         SetSizer(sizer.release());
+      }
+
+      void SetEffect(AudacityProject& project, const std::shared_ptr<Track>& track, RealtimeEffectState* state)
+      {
+         mProject = &project;
+         mTrack = track;
+         mEffectState = state;
+         auto desc = PluginManager::Get().GetPlugin(mEffectState->GetID());
+
+         mChangeButton->SetTranslatableLabel(desc->GetSymbol().Msgid());
+      }
+
+      void RemoveFromList()
+      {
+         if(mEffectState == nullptr)
+            return;
+         
+         auto effectName = mEffectState->GetEffect()->GetName();
+         AudioIO::Get()->RemoveState(*mProject, &*mTrack, *mEffectState);
+         ProjectHistory::Get(*mProject).PushState(
+            //i18n-hint: undo history, first parameter - realtime effect name, second - track name
+            XO("'%s' removed from '%s' effect stack").Format(effectName, mTrack->GetName()),
+            //i18n-hint: undo history record
+            XO("Remove realtime effect")
+         );
+      }
+      
+      void OnOptionsClicked(wxCommandEvent& event)
+      {
+         if(mEffectState == nullptr)
+            return;//not initialized
+         
+         const auto ID = mEffectState->GetID();
+         const auto effectPlugin = EffectManager::Get().GetEffect(ID);
+         if(effectPlugin == nullptr)
+         {
+            ///TODO: effect is not avaialble
+            ///should it be removed from the list on interaction or
+            ///during project load/plugin manager list modification?
```

Comment:
See `RealtimeEffectList::HandleXMLEndTag` for some answer to your question.  Yes, there is a check on success of `GetEffect` at the time the project is reopened, which we can assume, I think, follows all plugin initializations at startup.  And I think there is then no way to insert an effect from a plug-in that does not exist.

But can a plug-in ever be un-loaded while the session continues?  I don't think that ever happens.  If it did, perhaps then more uses of Observer and subscriptions might let the RealtimeEffectLists update themselves, and they would
 further propagate `Publish`-ed messages to these windows.

Reply:
>See RealtimeEffectList::HandleXMLEndTag for some answer to your question

But it happens silently, without any warnings, isn't it?

>And I think there is then no way to insert an effect from a plug-in that does not exist.

If there was a warning, user could have decided to close the project, install plugin and reopen this project again. Also, it's not necessary to remove `RealtimeEffectState` if plugin which it refers to is missing. There might be something that indicates a faulty plugin in the stack. But this is more UI/UX question I guess.